pkg/operator/mds/mds.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mds
+
+import (
+	"fmt"
+
+	"github.com/rook/rook/pkg/cephmgr/client"
+	cephmds "github.com/rook/rook/pkg/cephmgr/mds"
+	"github.com/rook/rook/pkg/cephmgr/mon"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	k8smon "github.com/rook/rook/pkg/operator/mon"
+	"k8s.io/client-go/1.5/kubernetes"
+	"k8s.io/client-go/1.5/pkg/api/v1"
+	extensions "k8s.io/client-go/1.5/pkg/apis/extensions/v1beta1"
+)
+
+const (
+	appName            = "mds"
+	dataPoolSuffix     = "-data"
+	metadataPoolSuffix = "-metadata"
+	keyringName        = "keyring"
+)
+
+type Cluster struct {
+	Namespace string
+	Version   string
+	Replicas  int32
+	factory   client.ConnectionFactory
+}
+
+func New(namespace, version string, factory client.ConnectionFactory) *Cluster {
+	return &Cluster{
+		Namespace: namespace,
+		Version:   version,
+		Replicas:  1,
+		factory:   factory,
+	}
+}
+
+func (c *Cluster) Start(clientset *kubernetes.Clientset, cluster *mon.ClusterInfo) error {
+	logger.Infof("start running mds")
+
+	if cluster == nil || len(cluster.Monitors) == 0 {
+		return fmt.Errorf("missing mons to start mds")
+	}
+
+	context := &clusterd.DaemonContext{ConfigDir: "/var/lib/rook"}
```

Comment:
when running inside kubernetes we have a defined directory we can count on in the pod. That said, i should make that more clear with a const and comments